### DISCUSSION TITLE
Option for voice to text button to switch between Google VTT or a third party such as Futo voice input.

### DISCUSSION THREAD

**Discussion by SuperSentient** [AnySoftKeyboard/AnySoftKeyboard#4019] [Ideas]
It would be nice if by pressing and hold the VTT button you could switch between Google VTT or something like Futo voice input.

**Comment by supertiz** [comment]
I have developed a version of ASK that supports offline voice recognition, with the help of vosk library.
https://github.com/supertiz/AnySoftKeyboard/tree/offlineVoiceRecognition
What do you think?

**Reply by menny** [reply]
That's cool. Can you share a screen recording of the UX?

**Reply by supertiz** [reply]
Here you are, from my phone.
Some minor things need to be fixed but it works!

https://github.com/user-attachments/assets/a0aca0c8-6ed2-4966-9bfe-3f694587b568
